### PR TITLE
Add Builder to ZKDistributedNonblockingLock to enable ZooScalability use

### LINKED ISSUE #1408: Make helix-lock module multi-ZK compliant
**Is your feature request related to a problem? Please describe.**
ZKDistributedNonblockingLock and its related helix-lock API were not ZooScalability-compliant, meaning they could not be used in a multi-ZK environment. 

**Describe the solution you'd like**
A PR that updates the said APIs so that we could use them in multi-ZK environment by adding a Builder that allows users to set multi-ZK parameters.

**Additional context**

### PR DESCRIPTION
**Pull request by narendly**
### Issues

- [x] My PR addresses the following Helix issues and references them in the PR description:

Fixes #1408

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

ZKDistributedNonblockingLock and its related helix-lock API were not ZooScalability-compliant, meaning they could not be used in a multi-ZK environment. This PR updates the said APIs so that we could use them in multi-ZK environment by adding a Builder that allows users to set multi-ZK parameters.

### Tests

- [x] The following tests are written for this issue:

No logic change.

- [x] The following is the result of the "mvn test" command on the appropriate module:

helix-lock:
```
[INFO] Tests run: 6, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.565 s - in TestSuite
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 6, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
```

### Commits

- My commits all reference appropriate Apache Helix GitHub issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Code Quality

- My diff has been formatted using helix-style.xml 
(helix-style-intellij.xml if IntelliJ IDE is used)

### CODE REVIEW

**Comment by mgao0** [comment]
Looks good in general. One thing to note: right now you are using the lock path that user passes in directly in the builder. But in the long run, we want to use the LockScope because we would need to support complex lock hierarchies. Is there a way you change the lock path in the builder to lock scope?

File: helix-lock/src/main/java/org/apache/helix/lock/helix/ZKDistributedNonblockingLock.java [hunk 1]
```diff
@@ -24,17 +24,21 @@
...
 import org.apache.helix.manager.zk.ZkBaseDataAccessor;
 import org.apache.helix.zookeeper.datamodel.ZNRecord;
 import org.apache.helix.zookeeper.zkclient.DataUpdater;
 import org.apache.log4j.Logger;
 
 
 /**
- * Helix nonblocking lock implementation based on Zookeeper
+ * Helix nonblocking lock implementation based on Zookeeper.
+ * NOTE: do NOT use ephemeral nodes in this implementation because ephemeral mode is not supported
```
Comment (zhangmeng916) [inline helix-lock/src/main/java/org/apache/helix/lock/helix/ZKDistributedNonblockingLock.java:40]: I'm not sure the purpose of this comment. Is this targeting whoever changes this class? what about non zooscalablity use case?

File: helix-lock/src/main/java/org/apache/helix/lock/helix/ZKDistributedNonblockingLock.java [hunk 2]
```diff
@@ -143,11 +147,65 @@ public ZNRecord update(ZNRecord current) {
...
+      return this;
+    }
+
+    public Builder setLockMsg(String lockMsg) {
+      _lockMsg = lockMsg;
+      return this;
+    }
+
+    public ZKDistributedNonblockingLock build() {
+      // Resolve which way we want to create BaseDataAccessor instance
```
Comment (zhangmeng916) [inline helix-lock/src/main/java/org/apache/helix/lock/helix/ZKDistributedNonblockingLock.java:193]: This line can be put together with the following comments.

File: helix-lock/src/main/java/org/apache/helix/lock/helix/ZKDistributedNonblockingLock.java (hunk 1 above)
Comment (narendly) [reply helix-lock/src/main/java/org/apache/helix/lock/helix/ZKDistributedNonblockingLock.java:40]: Yes, in order to keep things compatible, we shouldn't be using ephemeral nodes for the implementation of this class. If ephemeral nodes are needed, that would have to be a separate class with a different metadata store or mode of access.

**Review by mgao0 — APPROVED** [review]

**Comment by narendly** [comment]
This PR is ready to be merged.

**Comment by narendly** [comment]
Add Builder to ZKDistributedNonblockingLock to enable ZooScalability use

ZKDistributedNonblockingLock and its related helix-lock API were not ZooScalability-compliant, meaning they could not be used in a multi-ZK environment. This PR updates the said APIs so that we could use them in multi-ZK environment by adding a Builder that allows users to set multi-ZK parameters.

**Pull request merged**